### ISSUE TITLE
Have no access to plots.flux or plots.power

### ISSUE THREAD

**Issue by mmnmjm** [kklmn/xrt#21]
If I use plots.flux or plots.power in my generator code flux or power values are always zero.
Only if I use plots.intensity it seems to work.
How intensity is defined? Is it a flux density?

**Comment by yxrmz** [comment]
Are you working with synchrotron sources or geometric ones? Flux and power only make sense for the real sources.

**Comment by mmnmjm** [comment]
I  use the geometric source. Can I use plots.intensity instead of plots.flux?

**Comment by yxrmz** [comment]
It depends on what result you expect to see. Plot.intensity is simply the sum of individual intensities of all good rays in the beam you plot. Ray intensity equals 1 in the moment of generation and gets adjusted regarding the reflectivity/transmissivity of the materials of the optical elements on the path. Some rays may get lost on apertures or falling outside of the OE, these are marked as bad and do not contribute to the intensity sum. So, "intensity" is always related to the initial distribution, whereas "flux" and "power" in photons/s and watts are absolute values (in the given spectral boundaries, of course).

**Comment by mmnmjm** [comment]
I am interested to quantify the photon throughput. If I understand correctly plot.intensity is a measure for the number of photons that go through the setup in relation to the number of photons that are emitted from the source.

**Comment by kklmn** [comment]
> If I understand correctly plot.intensity is a measure for the number of photons that go through the setup in relation to the number of photons that are emitted from the source.

Yes, this is the number of good rays times the reflectivity/transmittivity at each optical surface. The name "intensity" is not the best for this quantity and should rather be "flux" but it was borrowed from Shadow.
